jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/InvoicePaymentResource.java

```diff
@@ -177,8 +182,30 @@ public Response createRefundWithAdjustments(final InvoicePaymentTransactionJson
             pluginProperties = extractPluginProperties(pluginPropertiesString);
         }
 
-        final Payment result = paymentApi.createRefundWithPaymentControl(account, payment.getId(), json.getAmount(), account.getCurrency(), transactionExternalKey,
-                                                                               pluginProperties, createInvoicePaymentControlPluginApiPaymentOptions(false), callContext);
+        final Payment result;
+        PaymentMethod paymentMethod = null;
+        try {
+            paymentMethod = paymentApi.getPaymentMethodById(payment.getPaymentMethodId(), false, false, pluginProperties, callContext);
+        } catch (PaymentApiException e) {
+            log.warn("Payment method {} does not found", payment.getPaymentMethodId());
```

Comment:
Because we now have the extra query parameter `externalPayment`, I don't even think we should do this check anymore - just retrieve the payment method when `externalPayment` is false and don't catch the exception.